app.js

```diff
@@ -36,41 +38,109 @@ store.on("error", function (error) {
 };
   
+const csrf_options = {
+    protected_operations: ["PATCH"],
+    protected_content_types: ["application/json"],
+    development_mode:  !inProduction,
+    cookie: true,
+};
+
 // If the application is in production
 if (inProduction) {    
     app.set("trust proxy", 1); // trust first proxy       
 }
   
 // Use the session middleware with the defined parameters
 app.use(session(sessionParms));
+app.set("view engine", "ejs");// Set the view engine to ejs for rendering views
+app.use(bodyParser.urlencoded({ extended: true })); // Use body-parser middleware to parse incoming request bodies
+app.use(flash()); // Add the connect-flash middleware
+app.use(cookieParser(process.env.SESSION_SECRET));
+app.use(express.urlencoded({ extended: false }));
+// app.use(csurf());
+app.use(methodOverride('_method'));
+
+// Security middlewares
+app.use(helmet());
+app.use(xssClean());
+const limiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 100 // limit each IP to 100 requests per windowMs
+});
+app.use(limiter);
 
 // Add the passport middleware
-
 passportInit();
 app.use(passport.initialize());
 app.use(passport.session());
 
-// Add the connect-flash middleware
-app.use(require("connect-flash")());
 
-app.use(require("./middleware/storeLocals"));
-app.get("/", (req, res) => {
-    res.render("index");
+
+app.use(csrf());
+// Middleware to add the CSRF token to the response locals
+app.use((req, res, next) => {
+    res.locals.csrfToken = req.csrfToken();
+    next();
 });
-app.use("/sessions", require("./routes/sessionRoutes"));
 
-// Replace the app.get and app.post statements for the "/secretWord" routes
+// Middleware to add flash messages to the response locals
+app.use((req, res, next) => {
+    res.locals.info = req.flash("info");
+    res.locals.errors = req.flash("error");
+    next();
+  });
+
+  const storeLocals = require("./middleware/storeLocals");
+  app.use(storeLocals);
+  
+app.get('/jobs/new', (req, res) => {
+      res.render('newJob', { job: null, _csrf: req.csrfToken() });
+  });
+  
+app.get('/jobs', async function(req, res) {
+    try {
+        // Fetch jobs from the database...
+        let jobs = await Job.find(); 
+        res.render('jobs', { jobs: jobs, _csrf: req.csrfToken() });
+    } catch (error) {
+        console.error(error);
+        res.status(500).send('An error occurred while fetching jobs');
+    }
+});
+  
+  app.get('/job', (req, res) => {
+      res.render('job', { job: null }); // For adding a new job
+  });
+  
+  app.get('/job/edit/:id', async (req, res) => {
+      // Fetch the job with the given ID from your database
+      try {
+          const job = await Job.findById(req.params.id);
+          res.render('job', { job }); // For editing an existing job
+      } catch (error) {
+          console.error(error);
+          res.status(500).send('An error occurred while fetching the job');
+      }
+  });
```

Comment:
Requires change:
Why are these routes/controllers in your app.js file!?!? Please extract these lines into their  respective route and controller counterparts.